### ISSUE TITLE
Blank Manager from corrupted cache Revisited

### ISSUE THREAD

**Issue by webandflow** [modxcms/revolution#9667]
webandflow created Redmine issue ID 9667

This is very similar to Bug 7532, but still happening on a new install of 2.2.6.

Similar to 7532, a corrupted cache file (in this case: resource.cache.php in ../lexicon/en/core) resulted in "white screening" throughout the entire manager.  Immediately before, I had saved a TV that I had created before and attempted to load another that I had needed to edit.

Since I couldn't actually access the error log, I found the file.  The latest entry read:

[2013-03-19 11:56:10](ERROR @ /connectors/layout/modx.config.js.php) Error caching lexicon topic lexicon/en/core/resource

Removing resource.cache.php resulted in the manager working correctly.

As I've been working with 2.2.6 for the past few days, this has happened 5-6 times.

 I had posted a similar bug some time ago - bug 4728.  In the same way this time, the cache file seemed to have been written in such a way that the last half of the file appeared first and the first half of the file appended to the end.  See line 58 of the attached file.

**Comment by soopermuse** [comment]
I'm having this issue with 2.3.3.
Breaks manager on a daily basis, lexicon cache has to be manually deleted via FTP.
It also seems to break any links generated by PhpThumb, so all gallery images break too.  I have to refresh the cache after getting back into manager to fix that one.

**Comment by OptimusCrime** [comment]
Could be some sort of permission problem?

**Comment by waizen** [comment]
I'm experiencing this on multiple sites...at least it seems like the same thing. Most of the problems manifest themselves as white blank pages, but not always that drastic. The sites range from v 2.2.6 to v.2.3.3. The v.2.3.3 site problem manifested itself with a mysterious disappearance of the Wayfinder menu on the frontend. Clearing the cache from the control panel brought it back The site does not have PhpThumb. It also has both compress_css and compress_js turned off. The error log is clear. Permission is 755 on the core>cache directory.

I agree...this is a major problem.

**Comment by Halfnium** [comment]
Waizen et. al., examine your Web hosts' error logs.  When I encountered the problem that you describe, with a site configured as Waizen describes, I found that virtual memory allocated when running the Manager would reach 2GB, the maximum available.  This functional observation suggests that the Manager attempts to fire off in parallel a bunch of work involving memory allocation.  If the Web host is not speedy enough, those allocations stack up and result in failure.

Two alternative interpretations follow that hypothesis.  First, that MODX is fine but requires a good, stout host in order to produce correct results.  Second, that MODX ought to be designed for tolerance of weak hosts, so that it always produces proper results, just more quickly on some hosts than on others.  Take your pick.

**Comment by Mark-H** [comment]
MODX using 2gb of ram? I think my VPS has 1,5gb of ram and runs over a dozen MODX sites just fine, never had this issue that I can remember. 

The manager does fire off a fair few requests because it's loading stuff over AJAX, but unless you're allocating a lot of memory to each of those requests (which they really don't need - most of them just send a request to the database and return the response quite quickly) I don't see how you could fill up that much memory from browsing the manager... must be something more at play than just that.

**Comment by OptimusCrime** [comment]
One of our VPS has 1GB RAM running 10 or more MODX sites using Apache2 just fine.

The high RAM usage must be caused by either an error in your installations or invalid server setup.

**Comment by waizen** [comment]
Okay, so probably not RAM on the server. Any other ideas? I'm not a server expert so I'm no help here but are you all satisfied that the problem is originating on a server level?

**Comment by waizen** [comment]
Not sure if this has anything to do with anything, but I've noticed that all of these occurrences have happened (and I came in to work this morning to 5 of them today) on servers that have cPanel on them. Our server with Direct Admin has never had a problem.

...Just throwing that out there.

**Comment by mrhaw** [comment]
Just experienced this issue with the above errors and blank manager page.

It only affected mgr home, so was able to access the manager by navigating to ?a=system/settings
I backed up and deleted the content of the cache folder manually. New cache generated without any progress.

Next I created an "empty" dashboard and that confirmed it was a widget giving me problems.
Then I created a new dashboard called "test" where I added back widgets one by one. 

MODX Security Feed widget was the culprit!

**Comment by opengeek** [comment]
@mrhaw what is the problem with the security feed widget? this is tangental to the issue here, at best and is not the cause of the original posted issue

**Comment by mrhaw** [comment]
I came to this thread after searching for the error messages displayed above.
Disabling the widget did not only solve the blank manager issue mentioned in this thread, but 
also made the error go away. I have not dived into what is exact cause. But RSS very often comes with 
special chars, different encoding and missing tags. In my book might help someone who navigate over here.

**Comment by opengeek** [comment]
@mrhaw This was due to us forcing requests to http; on systems with open_base_dir restrictions in effect, curl will not follow redirects—I had forgotten the feeds would be affected. This has been reverted, so the forum URLs work on http and https and do not attempt to redirect.

**Comment by mrhaw** [comment]
Another mystery solved!:+1:

**Comment by vierkantemeter** [comment]
I've also got this cache corruption problem with my install. I'm getting this kind of errors:

`(ERROR @ domain.eu/public_html/core/model/modx/modscript.class.php : 70) PHP warning: include(): Failed opening 'domain.eu/public_html/core/cache/includes/elements/modsnippet/43.include.cache.php' for inclusion (include_path='.:/usr/local/share/pear:/www/scripts/webshop:/www/scripts/public')`

Every time it's another snippet, so it's not the snippet itself. It really driving me nuts.. I don't know what to do about it. Should I move to another host?

- MODX 2.3.5
- Webhost: argeweb.nl (php 5.4.16)
- $config_options = array ('use_flock' => false,);
- If you want more server details, let me know

**Comment by pbowyer** [comment]
To those experiencing cache corruption issues: is your MODX install on an NFS drive? This was the cause of my corruption, which I experienced when using a Vagrant VM to develop.

For me, the fix was to set the MODX option (in the `modx_system_settings` table) with key `use_flock` and value `0`.

**Comment by OptimusCrime** [comment]
Closing this as the issue is too general and it caused by multiple underlying problems. Also, the issue with a blank manager due to compression errors has been fixed a while back.

Feel free to comment or reopen this issue if you disagree. Also, if you have a problem with a blank manager, feel free to open a new issue with the underlying problem identified.

@modxbot close
